lib/rouge/lexers/javascript.rb

```diff
@@ -177,6 +177,13 @@ def self.id_regex
           push :expr_start
         end
 
+        rule %r/(function)((?:\s|\\\s)+)/ do
+          groups Keyword, Text
+          push :funcname
+        end
+
+        rule %r/([a-z_]\w*)[ \t]*(?=(\(.*\)))/m, Name::Function
```

Comment:
Yes...but not with `id`. The issue is the best way for `rouge` to determine if the token is a function is by checking to see if it starts with a lowercase letter. I know that this isn't actually required by `javascript`, but otherwise it would match classes as functions as well and we don't want that. I do have something that I am testing that I think will work though. I'll push it shortly to let you see what you think.